core/src/main/java/com/linecorp/armeria/internal/metric/CaffeineMetricSupport.java

```diff
@@ -47,63 +59,145 @@ public static void setup(MeterRegistry registry, MeterId id, Cache<?, ?> cache)
 
     public static void setup(MeterRegistry registry, MeterId id, Cache<?, ?> cache, Ticker ticker) {
         final CaffeineMetrics metrics = MicrometerUtil.register(
-                registry, id, CaffeineMetrics.class, (r, i) -> new CaffeineMetrics(r, i, cache, ticker));
-        checkState(metrics.cache == cache, "Meters for a different Cache have been registered for id: %s", id);
+                registry, id, CaffeineMetrics.class, CaffeineMetrics::new);
+        metrics.add(cache, ticker);
     }
 
     private CaffeineMetricSupport() {}
 
+    enum Type {
+        HIT_COUNT,
+        MISS_COUNT,
+        EVICTION_COUNT,
+        EVICTION_WEIGHT,
+        LOAD_SUCCESS_COUNT,
+        LOAD_FAILURE_COUNT,
+        TOTAL_LOAD_TIME;
+
+        static final int count = Type.values().length;
```

Comment:
Just small note, this was confusing to me at first since some of the enum are also named count.

Reply:
Any suggestions for a better name? I used length first and then count. Didn't like any :-)

Reply:
Since it's only used once, I would just remove the field.